packages/wdio-browserstack-service/src/service.js

```diff
@@ -36,16 +44,18 @@ export default class BrowserstackService {
             this.sessionBaseUrl = 'https://api-cloud.browserstack.com/app-automate/sessions'
         }
 
+        this.scenariosThatRan = []
+
         return this._printSessionURL()
     }
 
-    afterSuite(suite) {
-        if (Object.prototype.hasOwnProperty.call(suite, 'error')) {
-            this.failures++
-        }
+    beforeFeature(uri, feature) {
+        const oldTitle = this.fullTitle
+        this.fullTitle = [this.fullTitle, feature.document.feature.name].filter(Boolean).join(', ')
```

Comment:
`.filter(Boolean)` why do this?

Reply:
Was under the impression we could run multiple features (spec files) on one session and this would correctly set the session name accordingly.


`[undefined, 'Feature 1'].filter(Boolean).join(', ')`
`['Feature 1', 'Feature 2'].filter(Boolean).join(', ')`
`['Feature 1, Feature 2', 'Feature 3'].filter(Boolean).join(', ')`

Reply:
So that means this.fullTitle can be undefined? If so, simply don't add it as it would make it explicit what you are trying to prevent from happening instead of patching it afterwards.

This would also mean that this will update with undefined:
```js
this._update(this.sessionId, { name: this.fullTitle }) // name can be undefined
```

Reply:
1 session equals 1 Feature so I've removed this